onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2726,15 +2726,73 @@ def aten_index_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::index_put")
 def aten_index_put(
-    self: TensorType,
-    indices: Optional[Sequence[TensorType]],
-    values: TensorType,
+    self: TReal,
+    indices: Sequence[TReal],
+    values: TReal,
     accumulate: bool = False,
-) -> TensorType:
+) -> TReal:
     """index_put(Tensor self, Tensor?[] indices, Tensor values, bool accumulate=False) -> Tensor"""
```

Comment:
Will indices be empty?

Reply:
if indices is empty, this function will be meaningless. PyTorch will return error if it is empty.